_posts/2021-05-17-changelog-147.markdown

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #147"
+date: 2021-05-17 12:00:00 +0300
+---
+
+
+In this release, we enable the new name resolution engine by default. Since this fundamental change has been made, we decided to increase the version of our plugin from `0.3` to `0.4`. So, meet the `IntelliJ Rust 0.4`!
+
+## New Features
+
+* [#7140] Enable [new name resolution engine](https://blog.jetbrains.com/clion/2020/12/intellij-rust-updates-for-2020-3/#new-resolve) by default. Our new algorithm uses the fixed-point iteration approach and aims to provide better support for complex glob imports (`use foo::*`) and re-exports (`use foo as bar`). Here is list of improvements offered by new engine:
+
+    * Properly support [textual scoping](https://doc.rust-lang.org/reference/macros-by-example.html#textual-scope) when resolving macros
+    * Resolve of multi-segment macro paths and macro calls inside files included by `include!`
+    * Resolve of [macros 2.0](https://github.com/rust-lang/rfcs/blob/master/text/1584-macros.md)
+    * Support file-level `#![macro_use]` and `cfg` attributes
+    * Find all possible `impl` blocks for a type alias. Previously, for performance reasons, they were processed only when the number of aliases for the corresponding type didn’t exceed 10. Now the crates that heavily use type aliases, such as [nalgebra](https://crates.io/crates/nalgebra) or [winapi](https://crates.io/crates/winapi), can be resolved properly
+
+  Note that the new algorithm works for top-level items only, so local variables and type inference functionality won’t be affected. If you run into an issue with the new engine, you can switch it back to the old one in the plugin settings `Settings | Languages & Frameworks | Rust | Use new name resolution engine`
+
+* [#6456] Add initial [Windows Subsystem for Linux (WSL)](https://www.jetbrains.com/help/idea/how-to-use-wsl-development-environment-in-product.html) support. It is available since 2021.1, and can be enabled using `org.rust.wsl` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features). Note that if there are both local Windows and WSL toolchains, the local one will be selected automatically. You can manually specify `\\wsl$` paths in `Settings | Languages & Frameworks | Rust` to use WSL toolchain, and we plan to improve it in the future
```

Comment:
```suggestion
* [#6456] Add initial [Windows Subsystem for Linux (WSL)](https://www.jetbrains.com/help/idea/how-to-use-wsl-development-environment-in-product.html) support. It is available since 2021.1, and can be enabled using `org.rust.wsl` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features). Note that if there are both local Windows and WSL toolchains, the local one will be selected automatically. You can manually specify desired WSL toolchain location with `\\wsl$` path in `Settings | Languages & Frameworks | Rust` settings. We plan to improve toolchain selection in the future
```